### PR TITLE
Feature/bl 8022 elastic search

### PR DESCRIPTION
**Pull request by ddiamondKainos**

### CODE REVIEW

File: docs/step-definitions.md [hunk 1]
```diff
@@ -279,4 +279,9 @@ These steps are specific to the DVSA MOT application screens.
...
 * I close extra tabs
 * I click the accordion section with the id ```..text..```
-* I click the ```..text..``` help link
\ No newline at end of file
+* I click the ```..text..``` help link
+
+### ElasticSearchStepDefinitons
+
+* I query elastic search with ```"..csv file name.."``` as ```"..data key.."```
+* I compare the search results for ```"..result set key.."``` and ```"..result set key.."```
```
Comment (andrew-lai-bjss) [inline docs/step-definitions.md]: Does not match method signature.

Also no entry for the new "I set the starting url..."

File: src/main/java/uk/gov/dvsa/mot/framework/WebDriverWrapper.java [hunk 2]
```diff
@@ -682,6 +688,7 @@ public String getElementText(String id) {
      * @return The text
      */
     public String getElementText(String startTag, String startText, String relativeXPath) {
+        waitForFullPageLoad();
```
Comment (andrew-lai-bjss) [inline src/main/java/uk/gov/dvsa/mot/framework/WebDriverWrapper.java]: Is this needed? 

Have you run the full suite to ensure it doesn't impact anywhere else that uses this?

File: src/main/java/uk/gov/dvsa/mot/framework/excel/ExcelDocument.java [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+                updatedValue.setCellValue(result2);
+
+                if (!elasticSearchResult1.getTopTenList().contains(result2)) {
+                    updatedValue.setCellStyle(addativeStyle);
+                }
+            }
+
+            rowNumber = Math.max(updatedRow, originalRow);
+            rowNumber++;
```
Comment (andrew-lai-bjss) [inline src/main/java/uk/gov/dvsa/mot/framework/excel/ExcelDocument.java]: Isn't this just the same as
rowNumber = Math.max(updatedRow, originalRow) + 1;

File: src/main/java/uk/gov/dvsa/mot/fixtures/mot/ElasticSearchStepDefinitions.java [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    private ArrayList<CSVRecord> processCsvDataForDefects(String csvFile) {
+        ArrayList<CSVRecord> records = new ArrayList<>();
+
+        File defectsData = new File("src/main/java/uk/gov/dvsa/mot/framework/elasticsearch/" + csvFile + ".csv");
+
+        try {
+            CSVParser parser = CSVParser.parse(defectsData, Charset.defaultCharset(), CSVFormat.EXCEL);
+            records.addAll(parser.getRecords());
+        } catch (IOException exception) {
+            logger.error(exception.getMessage());
```
Comment (andrew-lai-bjss) [inline src/main/java/uk/gov/dvsa/mot/fixtures/mot/ElasticSearchStepDefinitions.java]: Shouldn't really do this as you can lost the stack trace better to:
logger.error("Unable to parse CSV file", exception);

File: src/main/java/uk/gov/dvsa/mot/framework/excel/ExcelDocument.java [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+            indexCounter++;
+        }
+
+        try {
+            FileOutputStream fileOut = new FileOutputStream("build/reports/selenium/" + csvFile + "Results.xlsx");
+            workbook.write(fileOut);
+            fileOut.close();
+        } catch (java.io.IOException exception) {
+            logger.error(exception.getMessage());
```
Comment (andrew-lai-bjss) [inline src/main/java/uk/gov/dvsa/mot/framework/excel/ExcelDocument.java]: See earlier comment about logging errors.

File: docs/step-definitions.md (hunk 1 above)
Comment (ddiamondKainos) [reply docs/step-definitions.md]: changed the compare step and added the starting url

File: src/main/java/uk/gov/dvsa/mot/framework/WebDriverWrapper.java (hunk 2 above)
Comment (ddiamondKainos) [reply src/main/java/uk/gov/dvsa/mot/framework/WebDriverWrapper.java]: no longer needed, removed.

File: src/main/java/uk/gov/dvsa/mot/framework/excel/ExcelDocument.java (hunk 3 above)
Comment (ddiamondKainos) [reply src/main/java/uk/gov/dvsa/mot/framework/excel/ExcelDocument.java]: squashed to one line

File: src/main/java/uk/gov/dvsa/mot/fixtures/mot/ElasticSearchStepDefinitions.java (hunk 4 above)
Comment (ddiamondKainos) [reply src/main/java/uk/gov/dvsa/mot/fixtures/mot/ElasticSearchStepDefinitions.java]: updated error message format

File: src/main/java/uk/gov/dvsa/mot/framework/excel/ExcelDocument.java (hunk 5 above)
Comment (ddiamondKainos) [reply src/main/java/uk/gov/dvsa/mot/framework/excel/ExcelDocument.java]: updated error message format

**Pull request merged**